gums-core/src/main/java/gov/bnl/gums/account/AccountPoolMapper.java

```diff
@@ -163,6 +169,9 @@ public String getType() {
 
     @Override
     public AccountInfo mapUser(GridUser user, boolean createIfDoesNotExist) {
+        if (groupName != null && !groupName.equals("")) {
+            return new AccountInfo(getDB().retrieveMapping(user.getCertificateDN()), groupName);
```

Comment:
Good!  I was basing that on the mapUser() in [ManualAccountMapper.java](https://github.com/opensciencegrid/gums/blob/master/gums-core/src/main/java/gov/bnl/gums/account/ManualAccountMapper.java#L98).  But if it works to return AccountInfo(account, groupName), great -- I'll give that a shot.

(That might also mean I can get rid of some of the extra retrieveMapping() stuff I had to add in AccountPoolMapperDB.java, LocalPersistenceFactory.java, MockAccountPoolMapperDB.java ...)